hw/ip/spi_host/doc/_index.md

```diff
@@ -146,32 +151,117 @@ A preliminary investigation of DTR transfer mode suggests that proper support fo
-    - The number of dummy `sck` cycles to wait before reading data, {{< regref "COMMAND_0.DUMMY_CYCLES_0" >}}
-    - The number of bytes to receive at the full Dual or Quad rate, {{< regref "COMMAND_0.RX_CNT_0" >}}
-4. Read the peripheral response from the {{< regref "RXDATA" >}} register.
-
-### Configuration options
-
-The {{< regref "CONFIGOPTS_0" >}} multi-register has one entry per `cs_n` line and holds clock configuration and timing settings which are specific to each peripheral.
-Once the {{< regref "CONFIGOPTS_0" >}} multi-register has been programed for each SPI peripheral device, the values can be left unchanged.
+Any particular SPI command can be thought of one or more *segments*, each with different speed (number of active `sd` lines), direction and length.
+For example a quad SPI read transaction consists of 4 segments:
+1. A single byte instruction transmitted at *standard* data rate
+2. A three or four byte address transmitted at *quad* data rate
+3. A number of dummy cycles (no data transmitted or received)
+4. The desired data, received by `spi_host` at *quad* data rate.
+
+During any transaction, software can issue multiple segment descriptions to the `SPI_HOST` IP to control for changes in speed or direction. 
+
+Issuing a command then consists of the following steps:
+1. Configure the IP to be compatible with each attached peripheral.
+The {{< regref "CONFIGOPTS_0" >}} multi-register holds separate sets configuration settings, one for each `csb` line.
+In principle configuration of these device-specific options only needs to be once at initialization.
+2. Load the TX FIFO with the instructions and data to be transmitted to the remote device by writing to the {{< regref "DATA" >}} memory window.
+3. Specify which device should receive the next command using the {{< regref "CSID" >}} register.
+4. Wait for {{< regref "STATUS.READY" >}} before continuing.
+5. Issue speed, direction, and length details for the next command segment using the {{< regref "COMMAND" >}} register.
+If a command consists of multiple segments, then leave {{< regref "COMMAND.CSAAT" >}} (Chip-select active after transaction) for all segments except the last one.
+Setting {{< regref "COMMAND.CSAAT" >}} to 0 indicates the end of a transaction, prompting the IP to raise `csb` at the end of the segment. 
+6. Repeat steps 4 and 5 until all segments have been described.
+7. Read any peripheral response data from the RX FIFO by reading from the {{< regref "DATA" >}} memory window.
+
+### About Command Segments
+
+The structure of a SPI command varies from device-to-device and from command-to-command.
+
+To support a variety of different I/O sequences the SPI_HOST FSM treats each command as a sequence of segments such each with a defined length, direction and speed.
+
+Starting with the simplest case of standard SPI devices, these commands for these devices are very consistent in structure: the host transmits data on `sd[0]`, and always receives data on `sd[1]`.
```

Comment:
```suggestion
In case of a standard SPI device the commands are very consistent in structure: the host transmits data on `cio_sd[0]`, and always receives data on `cio_sd[1]`.
```